web/src/test/java/org/cbioportal/web/SampleListControllerTest.java

```diff
@@ -0,0 +1,256 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebAppConfiguration
+@ContextConfiguration("/applicationContext-web.xml")
+@Configuration
+public class SampleListControllerTest {
+
+    private static final int TEST_LIST_ID_1 = 1;
+    private static final String TEST_STABLE_ID_1 = "test_stable_id_1";
+    private static final int TEST_CANCER_STUDY_ID_1 = 1;
+    private static final String TEST_STUDY_IDENTIFIER_1 = "test_study_identifier_1";
+    private static final String TEST_CATEGORY_1 = "test_datatype_1";
+    private static final String TEST_NAME_1 = "test_name_1";
+    private static final String TEST_DESCRIPTION_1 = "test_description_1";
+    private static final String TEST_CANCER_STUDY_IDENTIFIER_1 = "test_study_1";
+    private static final String TEST_TYPE_OF_CANCER_ID_1 = "test_type_of_cancer_id_1";
+    private static final String TEST_STUDY_NAME_1 = "test_study_name_1";
+    private static final String TEST_SHORT_NAME_1 = "test_short_name_1";
+    private static final String TEST_STUDY_DESCRIPTION_1 = "test_study_description_1";
+    private static final int TEST_LIST_ID_2 = 2;
+    private static final String TEST_STABLE_ID_2 = "test_stable_id_2";
+    private static final int TEST_CANCER_STUDY_ID_2 = 2;
+    private static final String TEST_STUDY_IDENTIFIER_2 = "test_study_identifier_2";
+    private static final String TEST_CATEGORY_2 = "test_datatype_2";
+    private static final String TEST_NAME_2 = "test_name_2";
+    private static final String TEST_DESCRIPTION_2 = "test_description_2";
+    public static final String TEST_SAMPLE_ID_1 = "test_sample_id_1";
+    public static final String TEST_SAMPLE_ID_2 = "test_sample_id_2";
+
+    @Autowired
+    private WebApplicationContext wac;
+
+    @Autowired
+    private SampleListService sampleListService;
+    private MockMvc mockMvc;
+
+    @Bean
+    public SampleListService sampleListService() {
+        return Mockito.mock(SampleListService.class);
+    }
+
+    @Before
+    public void setUp() throws Exception {
+
+        Mockito.reset(sampleListService);
+        mockMvc = MockMvcBuilders.webAppContextSetup(wac).build();
+    }
+    
+    @Test
+    public void getAllSampleListsDefaultProjection() throws Exception {
```

Comment:
Add web layer test for getAllSampleListsDetailedProjection() and getAllSampleListsInStudyDetailedProjection()